Tests/SonarQube.TeamBuild.PreProcessor.Tests/SonarWebServiceTest.cs

```diff
@@ -136,20 +147,49 @@ public void GetInternalKeys()
             expected["fxcop:UriParametersShouldNotBeStrings"] = "CA1054";
             var actual = ws.GetInternalKeys("fxcop");
 
-            Assert.AreEqual(true, expected.Count == actual.Count && !expected.Except(actual).Any());
+            // default
+            var expected1 = new Dictionary<string, string>();
+            expected1["sonar.property1"] = "value1";
+            expected1["sonar.property2"] = "value2";
+            expected1["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            var actual1 = ws.GetProperties("foo bar", null);
+
+            Assert.AreEqual(true, expected1.Count == actual1.Count && !expected1.Except(actual1).Any());
+
+            // branch specific
+            var expected2 = new Dictionary<string, string>();
+            expected2["sonar.property1"] = "anotherValue1";
+            expected2["sonar.property2"] = "anotherValue2";
+            expected2["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            var actual2 = ws.GetProperties("foo bar", "aBranch");
+
+            Assert.AreEqual(true, expected2.Count == actual2.Count && !expected2.Except(actual2).Any());
         }
 
         [TestMethod]
         public void GetProperties()
         {
+            // Check that properties are correctly defaulted as well as branch-specific
             downloader.Pages["http://myhost:222/api/properties?resource=foo+bar"] = "[{\"key\": \"sonar.property1\",\"value\": \"value1\"},{\"key\": \"sonar.property2\",\"value\": \"value2\"}]";
-            var expected = new Dictionary<string, string>();
-            expected["sonar.property1"] = "value1";
-            expected["sonar.property2"] = "value2";
-            expected["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
-            var actual = ws.GetProperties("foo bar");
+            downloader.Pages["http://myhost:222/api/properties?resource=foo+bar%3AaBranch"] = "[{\"key\": \"sonar.property1\",\"value\": \"anotherValue1\"},{\"key\": \"sonar.property2\",\"value\": \"anotherValue2\"}]";
 
-            Assert.AreEqual(true, expected.Count == actual.Count && !expected.Except(actual).Any());
+            // default
+            var expected1 = new Dictionary<string, string>();
+            expected1["sonar.property1"] = "value1";
+            expected1["sonar.property2"] = "value2";
+            expected1["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            var actual1 = ws.GetProperties("foo bar", null);
+
+            Assert.AreEqual(true, expected1.Count == actual1.Count && !expected1.Except(actual1).Any());
+
+            // branch specific
```

Comment:
Further offline discussion has led to reverting this test to the original form. It is not the responsibility of the SonarWebServiceTest class to check the data-missing cases as GetProperties is not designed to be safe in those cases.